modules/gateway/nodes.go

```diff
@@ -33,24 +35,32 @@ func (g *Gateway) addNode(addr modules.NetAddress) error {
 
 // pingNode verifies that there is a reachable node at the provided address
 // by performing the Sia gateway handshake protocol.
-func (g *Gateway) pingNode(addr modules.NetAddress) error {
+func (g *Gateway) pingNode(addr modules.NetAddress) (err error) {
 	// Ping the untrusted node to see whether or not there's actually a
 	// reachable node at the provided address.
 	conn, err := g.dial(addr)
 	if err != nil {
-		return err
+		return
 	}
 	defer conn.Close()
-	// If connection succeeds, supply an unacceptable version so that we
-	// will not be added as a peer.
-	//
-	// NOTE: this is a somewhat clunky way of specifying that you didn't
-	// actually want a connection.
-	_, err = connectVersionHandshake(conn, "0.0.0")
-	if err == errPeerRejectedConn {
-		err = nil // we expect this error
+
+	// Read the node's version.
+	remoteVersion, err := connectVersionHandshake(conn, build.Version)
+	if err != nil {
+		return
 	}
-	return err
+
+	if build.VersionCmp(remoteVersion, sessionHandshakeUpgradeVersion) < 0 {
+		return // for older versions, this is where pinging ends
```

Comment:
@lukechampine is this going to cause problems since the old peer is going to expect the connection to be maintained? We are not writing 0.0.0 as our version anymore, so we need to make sure that we hang up some other way.